### PR TITLE
[CU-86dtx0a3a] install sentry

### PR DESCRIPTION
**Pull request by cgarcia-lightit**
feat: implement sentry and resolve console error in the console to avoid unnecessary logs in sentry, also log information to provide more data case profiling failed

# ⚡ Install Sentry - [CU-86dtx0a3a](https://app.clickup.com/t/86dtx0a3a) ⚡

## 💻 What type of change is this?

- [x] 💎 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Styling
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build / CI

## ⭐ Description

For more background, see ticket **CU-86dtx0a3a[in-progress].**

### Before

image

### After

image

### To review
- [ ] I have tested this change locally in multiple screen sizes
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If my task include an endpoint, I add the endpoint to Hopscotch/Postman Project
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by martinopp** [comment]
Task linked: [CU-86dtx0a3a Install sentry in frontend (react and llm)](https://app.clickup.com/t/86dtx0a3a)

File: apps/eo_web/src/components/HowEOWorks.tsx [hunk 1]
```diff
@@ -37,9 +36,9 @@ export const HowEOWorks = ({ flow }: HowEOWorksProps) => {
           : "How EO Care Plans Work"}
       </Typography>
       <EoCarousel>
-        {carouselItems.map(({ title, content, step, icon: Icon }) => (
+        {carouselItems.map(({ title, content, step, icon: Icon }, index) => (
           <article
-            key={step}
+            key={`${title}-${step}-${index}`}
```
Comment (sgoycoechea-lightit) [inline apps/eo_web/src/components/HowEOWorks.tsx:41]: I think adding the index to the key isn't needed, the title and step is enough, for all of them to be different, right?

File: apps/eo_web/src/components/Carousel.tsx [hunk 2]
```diff
@@ -59,11 +56,9 @@ export const EoCarousel = ({ children }: EoCarouselProps) => {
         >
           {children.map((item, key) => {
             return (
-              <Fragment key={`${key}`}>
-                <SwiperSlide>
-                  <div className="mb-12">{item}</div>
-                </SwiperSlide>
-              </Fragment>
+              <SwiperSlide key={`${id}-${key}`}>
```
Comment (sgoycoechea-lightit) [inline apps/eo_web/src/components/Carousel.tsx:59]: `useId` shouldn't be used for keys:

https://react.dev/reference/react/useId#caveats

File: apps/eo_web/src/components/FAQs.tsx [hunk 3]
```diff
@@ -52,7 +49,7 @@ export const FAQs = ({ flow }: FAQsProps) => {
         </Typography>
         <div className="flex flex-col gap-6">
           {faqList.map(({ title, content }) => (
-            <Collapsible key={title} title={title} active={false}>
+            <Collapsible key={`${title}-${faqId}`} title={title} active={false}>
```
Comment (sgoycoechea-lightit) [inline apps/eo_web/src/components/FAQs.tsx:52]: Same here about using `useId`, and I think the title is enough for the key

File: apps/eo_web/src/helpers/sentry.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import { env } from "~/env";
+
+
+const sentryDns =
+  window.data.getEnv("VITE_SENTRY_DSN_PUBLIC") ?? env.VITE_SENTRY_DSN_PUBLIC;
+
+if (sentryDns) {
+  Sentry.init({
+    environment: window.data.getEnv("VITE_ENV") ?? env.VITE_APP_ENV,
```
Comment (sgoycoechea-lightit) [inline apps/eo_web/src/helpers/sentry.ts:18]: What if the VITE_APP_ENV is not set?

Also shouldn't it be like this?:

```suggestion
    environment: window.data.getEnv("VITE_APP_ENV") ?? env.VITE_APP_ENV,
```

**Pull request merged**